woocommerce/woocommerce-ios
# [Order Creation] Prevent line deletion via steppers

Closes: #11650


## Description


Up to now, it’s been possible to remove a product by tapping the delete button in the product stepper. We previously made this easier to identify by changing the button to a red `x`, however it was still easy to accidentally trigger, and annoying to recover from (as the product needs to be found and added again.)

As one attempt to improve this, we tried adding an undo button in a notice after deletion, but it opened up too many issues of unreliability, e.g. sometimes adding the same product as two lines of the order.

Since there’s a dedicated `remove product from order` button, visible alongside the stepper, we’ve decided to remove this functionality from the stepper.

This PR disables the `-` button of the stepper when it’s at the minimum quantity.

## Testing instructions


1. Launch the app
2. Navigate to `Orders > +`
3. Add a product
4. Observe that the `-` button on the stepper is disabled (visually and taps do not work)
5. Tap `+`
6. Observe that the `-` button is enabled
7. Tap `-`
8. Observe that the `-` button is disabled again.

Repeat the above test while editing an order.

Note that the button no longer is an `x` and no longer deletes the line item. The `Remove item from order` button still works and is the way to remove a line item.

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/2472348/574d51db-abac-4b5b-8f26-82b8f310e49e


---
